src/ILCompiler.Compiler/src/Compiler/LibraryRootProvider.cs

```diff
@@ -12,13 +15,25 @@ namespace ILCompiler
     /// </summary>
     public class LibraryRootProvider : ICompilationRootProvider
     {
+        /// <summary>
+        /// Symbolic name under which the managed entrypoint is exported.
+        /// </summary>
+        public const string ManagedEntryPointMethodName = "__managed__Startup";
+
         private EcmaModule _module;
+        private IList<MethodDesc> _libraryInitializers;
 
         public LibraryRootProvider(EcmaModule module)
         {
             _module = module;
         }
 
+        public LibraryRootProvider(EcmaModule module, IList<MethodDesc> libraryInitializers)
```

Comment:
For this we should just create a separate `ICompilationRootProvider` instead (e.g. a `NativeLibraryInitializerRootProvider` that looks similar to the existing `MainMethodRootProvider`).

`LibraryRootProvider` would go over all the methods and types in the provided module and root them for compilation. What you need here is just a root for the Initialize method (the new root provider will do that) and a root for the `NativeCallable` methods (existing `ExportedMethodsRootProvider` does that). You'll get natural tree shaking that way (things that are unreachable don't get compiled).